distributed/scheduler.py

```diff
@@ -1751,7 +1751,11 @@ def _to_dict(self, *, exclude: Container[str] = None):
             exclude = set()
         members = inspect.getmembers(self)
         return recursive_to_dict(
-            {k: v for k, v in members if k not in exclude and not callable(v)},
+            {
+                k: v
+                for k, v in members
+                if not k.startswith("__") and k not in exclude and not callable(v)
```

Comment:
I'm sure there's a good reason behind this addition, but it's not clear to me where the leading `__` is coming from

Reply:
See https://github.com/dask/distributed/issues/5522

> I accidentally found a different infinite recursion path on the scheduler, where `scheduler.TaskState` traversed into `__annotations__`, which somehow created some circular references.